src/BenchmarkDotNet/Diagnosers/EventPipeProfiler.cs

```diff
@@ -0,0 +1,235 @@
+            Path.GetDirectoryName(fileName).CreateIfNotExists();
+            benchmarkToTraceFile[parameters.BenchmarkCase] = fileName;
+
+            collectingTask = new Task(() =>
+            {
+                try
+                {
+                    using (var fs = new FileStream(fileName, FileMode.Create, FileAccess.Write))
+                    {
+                        var buffer = new byte[16 * 1024];
+
+                        while (true)
+                        {
+                            int nBytesRead = session.EventStream.Read(buffer, 0, buffer.Length);
+                            if (nBytesRead <= 0)
+                                break;
+                            fs.Write(buffer, 0, nBytesRead);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logger.WriteLine(LogKind.Error, $"An exception occurred during reading trace stream: {ex}");
+                }
+            });
+            collectingTask.Start();
+        }
+
+        public IEnumerable<Metric> ProcessResults(DiagnoserResults results)
+        {
+            Task.WaitAll(collectingTask);
+
+            if (benchmarkToTraceFile.TryGetValue(results.BenchmarkCase, out var traceFilePath))
+                ConvertToSpeedscope(results.BenchmarkCase, traceFilePath);
+
+            return Array.Empty<Metric>();
+        }
+        
+        public void DisplayResults(ILogger resultLogger)
+        {
+            resultLogger.WriteLine();
+            resultLogger.WriteLineHeader(LogSeparator);
+
+            if (logger.CapturedOutput.Any())
+            {
+                foreach (var line in logger.CapturedOutput)
+                    resultLogger.Write(line.Kind, line.Text);
+
+                resultLogger.WriteLine();
+            }
+            if (!benchmarkToTraceFile.Any())
+                return;
+
+            resultLogger.WriteLineInfo($"Exported {benchmarkToTraceFile.Count} trace file(s). Example:");
+            resultLogger.WriteLineInfo(benchmarkToTraceFile.Values.First());
+
+            resultLogger.WriteLineHeader(LogSeparator);
+        }
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            foreach (var benchmark in validationParameters.Benchmarks)
+            {
+                var runtime = benchmark.Job.ResolveValue(EnvironmentMode.RuntimeCharacteristic, EnvironmentResolver.Instance);
+
+                if (runtime.RuntimeMoniker < RuntimeMoniker.NetCoreApp30)
+                {
+                    yield return new ValidationError(true, $"{nameof(EventPipeProfiler)} supports only .NET Core 3.0+", benchmark);
+                }
+            }
+        }
+
+        public string ShortName => "EventPipe";
+
+        private void ConvertToSpeedscope(BenchmarkCase benchmarkCase, string traceFilePath)
```

Comment:
this class is responsible for profiling using event pipe (`EventPipeProfiler`) could you please move everything that is related to exporting to speedscope to a dedicated type?